### ISSUE TITLE
await-promise: misses check 'for await (... of ...)'

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#3232]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
async function main(param: Array<Promise<number>>) {
    for await (const foo of param) {
        foo; // foo is of type Promise<number>, because 'for await' does not await each entry
        // 'for await' is only useful when iterating AsyncIterableIterator<T>
    }
}
```

with `tslint.json` configuration:

```json
"await-promise": true
```

#### Actual behavior
No warning
#### Expected behavior
Warning when using `for await` on anything that is not `AsyncIterableIterator<T>`
